utilities/backupable/backupable_db.cc

```diff
@@ -2142,106 +2141,118 @@ Status BackupEngineImpl::BackupMeta::LoadFromFile(
+                               FileOptions(), &backup_meta_reader, nullptr);
+    if (!s.ok()) {
+      return s;
+    }
   }
-  std::unique_ptr<char[]> buf(new char[max_backup_meta_file_size_ + 1]);
-  Slice data;
-  s = backup_meta_reader->Read(max_backup_meta_file_size_, &data, buf.get());
 
-  if (!s.ok() || data.size() == max_backup_meta_file_size_) {
-    return s.ok() ? Status::Corruption("File size too big") : s;
+  // Failures handled at the end
+  std::string line;
+  if (backup_meta_reader->ReadLine(&line)) {
+    timestamp_ = std::strtoull(line.c_str(), nullptr, /*base*/ 10);
   }
-  buf[data.size()] = 0;
-
-  uint32_t num_files = 0;
-  char *next;
-  timestamp_ = strtoull(data.data(), &next, 10);
-  data.remove_prefix(next - data.data() + 1); // +1 for '\n'
-  sequence_number_ = strtoull(data.data(), &next, 10);
-  data.remove_prefix(next - data.data() + 1); // +1 for '\n'
-
-  if (data.starts_with(kMetaDataPrefix)) {
-    // app metadata present
-    data.remove_prefix(kMetaDataPrefix.size());
-    Slice hex_encoded_metadata = GetSliceUntil(&data, '\n');
-    bool decode_success = hex_encoded_metadata.DecodeHex(&app_metadata_);
-    if (!decode_success) {
-      return Status::Corruption(
-          "Failed to decode stored hex encoded app metadata");
+  if (backup_meta_reader->ReadLine(&line)) {
+    sequence_number_ = std::strtoull(line.c_str(), nullptr, /*base*/ 10);
+  }
+  if (backup_meta_reader->ReadLine(&line)) {
+    Slice data = line;
+    if (data.starts_with(kMetaDataPrefix)) {
+      // app metadata present
+      data.remove_prefix(kMetaDataPrefix.size());
+      bool decode_success = data.DecodeHex(&app_metadata_);
+      if (!decode_success) {
+        return Status::Corruption(
+            "Failed to decode stored hex encoded app metadata");
+      }
+      line.clear();
+    } else {
+      // process the line below
     }
+  } else {
+    line.clear();
+  }
+  uint32_t num_files = UINT32_MAX;
+  if (!line.empty() || backup_meta_reader->ReadLine(&line)) {
+    num_files = static_cast<uint32_t>(strtoul(line.c_str(), nullptr, 10));
   }
-
-  num_files = static_cast<uint32_t>(strtoul(data.data(), &next, 10));
-  data.remove_prefix(next - data.data() + 1); // +1 for '\n'
-
   std::vector<std::shared_ptr<FileInfo>> files;
+  while (backup_meta_reader->ReadLine(&line)) {
+    std::vector<std::string> components = StringSplit(line, ' ');
 
-  // WART: The checksums are crc32c, not original crc32
-  Slice checksum_prefix("crc32 ");
+    if (components.size() < 1) {
+      return Status::Corruption("Empty line instead of file entry.");
+    }
 
-  for (uint32_t i = 0; s.ok() && i < num_files; ++i) {
-    auto line = GetSliceUntil(&data, '\n');
-    // filename is relative, i.e., shared/number.sst,
-    // shared_checksum/number.sst, or private/backup_id/number.sst
-    std::string filename = GetSliceUntil(&line, ' ').ToString();
+    const std::string& filename = components[0];
```

Comment:
Not supported. Sorry. That's the sort of thing we would consider if this were a schema change.

Reply:
Actually, I don't think RocksDB generates file names with spaces to be part of a DB under the DB dir, so because these are relative paths from the backup_dir, there are no spaces, even if the DB dir and/or backup dir have spaces.